version: '3.8'

services:
  postgres:
    image: postgres:latest
    container_name: postgres
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: database
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 5s
      timeout: 5s
      retries: 5
    volumes:
      - ./DB/create_tables.sql:/docker-entrypoint-initdb.d/create_tables.sql
    ports:
      - "5432:5432"

  api:
    build:
      context: .
    container_name: python
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
      - ./api:/app
    environment:
      DATABASE_URL: "postgresql+psycopg2://postgres:postgres@postgres:5432/database"
    command: sh -c "python app.py && tail -f /dev/null"
    ports:
      - "8086:8086"